tests/system/src/testapp7/java/mymod/myapp7/DataUrlWithModuleLayer.java

```diff
@@ -0,0 +1,115 @@
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package myapp7;
+
+import java.lang.module.ModuleDescriptor;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.scene.Scene;
+import javafx.scene.layout.BorderPane;
+import javafx.scene.web.WebView;
+import javafx.stage.Stage;
+
+public class DataUrlWithModuleLayer extends Application {
+
+    @Override
+    public void start(Stage primaryStage) throws Exception {
+        Module module = Application.class.getModule();
+
+        if (module == null) {
+            System.err.println("Failure: Module for Application not found");
+            System.exit(1);
+        }
+
+        if (! module.isNamed()) {
+            System.err.println("Failure: Expected named module");
+            System.exit(1);
+        }
+
+        ModuleDescriptor moduleDesc = module.getDescriptor();
+
+        if (moduleDesc.isAutomatic()) {
+            System.err.println("Failure: Automatic module found");
+            System.exit(1);
+        }
+
+        if (moduleDesc.isOpen()) {
+            System.err.println("Failure: Open module found");
+            System.exit(1);
+        }
+
+        BorderPane root = new BorderPane();
+        Scene scene = new Scene(root);
+
+        WebView webview = new WebView();
+        root.setCenter(webview);
+
+        // The title tag value is made available via the getTitle nethod. The
+        // JS code changes the value, so that the enclosing application can
+        // detect, that the JS code was run.
+        String checkJS = "document.getElementsByTagName(\"title\")[0].textContent='Executed'";
+        String checkJSEncoded = Base64.getEncoder().encodeToString(checkJS.getBytes(StandardCharsets.UTF_8));
+
+        String script = "<html>"
+            + "<head>"
+            + "<title>Armed</title>"
+            + "</head>"
+            + "<body>"
+            + "<h1>Test for loading a data URL</h1>"
+            + "<p>The test is successful, if the JVM does not crash with a SEGFAULT.</p>"
+            + "<script src=\"data:application/javascript;base64," + checkJSEncoded + "\"></script>"
+            + "</body>"
+            + "</html>";
+
+        webview.getEngine().loadContent(script);
+
+        primaryStage.setScene(scene);
+        primaryStage.setWidth(1024);
+        primaryStage.setHeight(768);
+        primaryStage.show();
+
+        new Thread() {
+            @Override
+            public void run() {
+                while (true) {
```

Comment:
Rather than a spin loop in a new thread, it's probably better to wait for the content to be loaded (checked using a listener on `WebEngine::getLoadWorker::stateProperty`) before doing the test, which you shouldn't need to do in a loop. In any case you will want a timeout so that if this process hangs it doesn't hang the test suite, which has happened in the past.